hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieAvroFileWriterFactory.java

```diff
@@ -51,7 +51,7 @@ protected HoodieFileWriter newParquetFileWriter(
       String instantTime, Path path, Configuration conf, HoodieConfig config, Schema schema,
       TaskContextSupplier taskContextSupplier) throws IOException {
     boolean populateMetaFields = config.getBooleanOrDefault(HoodieTableConfig.POPULATE_META_FIELDS);
-    boolean enableBloomFilter = populateMetaFields;
+    boolean enableBloomFilter = populateMetaFields && config.getBooleanOrDefault(HoodieStorageConfig.PARQUET_WITH_BLOOM_FILTER_ENABLED);
```

Comment:
Possible to add a check here to ensure that user is not using Bloom Index too?

Maybe put it into a function so that it is visible/can be used for UT.

Reply:
Nice advice, will adjust